src/olympia/reviewers/tests/test_views.py

```diff
@@ -5754,48 +5726,162 @@ def test_show_listed_and_unlisted_with_permissions(self):
             },
         ]
 
-    def test_draft_comment_patch_not_allowed(self):
-        url = reverse_ns('reviewers-versions-draft-comment', kwargs={
+    def test_draft_comment_create_and_retrieve(self):
+        user = UserProfile.objects.create(username='reviewer')
+        self.grant_permission(user, 'Addons:Review')
+        self.client.login_api(user)
+
+        data = {
+            'comment': 'Some really fancy comment',
+            'lineno': 20,
+            'filename': 'manifest.json',
+        }
+
+        url = reverse_ns('reviewers-versions-draft-comment-list', kwargs={
             'addon_pk': self.addon.pk,
-            'pk': self.version.pk
+            'version_pk': self.version.pk
         })
 
-        response = self.client.patch(url, {})
-        assert response.status_code == 405
+        response = self.client.post(url, data)
+        comment_id = response.json()['id']
+        assert response.status_code == 201
+
+        response = self.client.post(url, data)
+        assert response.status_code == 201
 
-    def test_draft_comment(self):
+        assert DraftComment.objects.count() == 2
+
+        response = self.client.get(url)
+
+        request = APIRequestFactory().get('/')
+        request.user = user
+
+        assert response.json()['count'] == 2
+        assert response.json()['results'][0] == {
+            'id': comment_id,
+            'filename': 'manifest.json',
+            'lineno': 20,
+            'comment': 'Some really fancy comment',
+            'version': json.loads(json.dumps(
+                VersionSerializer(self.version).data,
+                cls=amo.utils.AMOJSONEncoder)),
+            'user': json.loads(json.dumps(
+                BaseUserSerializer(
+                    user, context={'request': request}).data,
+                cls=amo.utils.AMOJSONEncoder))
```

Comment:
I don't suppose it matters, but I'm not sure you need to specifically use `AMOJSONEncoder` because `BaseUserSerializer` doesn't have any translations you'd need to flatten.

Reply:
it's certainly not needed here. I just chose to use it as we're using it everywhere in the code and usually don't explicitly import the regular `JSONEncoder` from Django.